pallets/asset/src/lib.rs

```diff
@@ -919,53 +873,79 @@ impl<T: Config> Module<T> {
         })
     }
 
     /// Accept and process a token ownership transfer.
-    fn base_accept_token_ownership_transfer(origin: T::RuntimeOrigin, id: u64) -> DispatchResult {
-        let to = Identity::<T>::ensure_perms(origin)?;
-        <Identity<T>>::accept_auth_with(&to.into(), id, |data, auth_by| {
-            let ticker = extract_auth!(data, TransferAssetOwnership(t));
+    fn base_accept_token_ownership_transfer(
+        origin: T::RuntimeOrigin,
+        auth_id: u64,
+    ) -> DispatchResult {
+        let caller_did = Identity::<T>::ensure_perms(origin)?;
+
+        <Identity<T>>::accept_auth_with(&caller_did.into(), auth_id, |auth_data, auth_by| {
+            let asset_id = extract_auth!(auth_data, TransferAssetOwnership(asset_id));
 
-            // Get the token details and ensure it exists.
-            let mut token = Self::token_details(&ticker)?;
+            let mut security_token = Self::try_get_security_token(&asset_id)?;
 
             // Ensure the authorization was created by a permissioned agent.
-            <ExternalAgents<T>>::ensure_agent_permissioned(ticker, auth_by)?;
-
-            // Get the ticker registration and ensure it exists.
-            let mut reg = Self::ticker_registration(&ticker).ok_or(Error::<T>::NoSuchAsset)?;
-            let old_owner = reg.owner;
-            AssetOwnershipRelations::remove(old_owner, ticker);
-            AssetOwnershipRelations::insert(to, ticker, AssetOwnershipRelation::AssetOwned);
-            // Update ticker registration.
-            reg.owner = to;
-            <Tickers<T>>::insert(&ticker, reg);
-            // Update token details.
-            token.owner_did = to;
-            Tokens::insert(&ticker, token);
-            Self::deposit_event(RawEvent::AssetOwnershipTransferred(to, ticker, old_owner));
+            <ExternalAgents<T>>::ensure_agent_permissioned(&asset_id, auth_by)?;
+
+            // If the asset is linked to a unique ticker, the ticker registration must be updated.
+            if let Some(ticker) = AssetIDTicker::get(&asset_id) {
+                UniqueTickerRegistration::<T>::try_mutate(
+                    &ticker,
+                    |ticker_registration| -> DispatchResult {
+                        match ticker_registration {
+                            Some(ticker_registration) => {
+                                // Removes the old owner
+                                TickersOwnedByUser::remove(ticker_registration.owner, ticker);
+                                // Adds the new owner
+                                TickersOwnedByUser::insert(caller_did, ticker, true);
+                                // Updates ticker registration
+                                ticker_registration.owner = caller_did;
+                                Ok(())
+                            }
+                            None => Err(Error::<T>::TickerRegistrationNotFound.into()),
+                        }
+                    },
+                )?;
```

Comment:
The `transfer_ticker` method will also emit an event for the ticker ownership change.
```suggestion
                let ticker_registration = UniqueTickerRegistration::<T>::try_get(&ticker)
                      .ok_or(Error::<T>::TickerRegistrationNotFound.into())?;
                Self::transfer_ticker(ticker_registration, ticker, caller_did);
```